### DISCUSSION TITLE
Autologin problems

### DISCUSSION THREAD

**Discussion by Valiumlord** [ltsp/ltsp#733] [Q&A]
Hello, im having trouble with LTSP autologin. I've setup LTSP server on proxmox and the networkboot works fine with my test machine except autologin in. Accounts that people log in come over NFS that works fine as well with manual login. Maybe im missing something and would love some help.
I'll post my ltsp.conf

`[clients]
FSTAB_2="xxx.xx.x.xxx/home    /home   nfs    vers=4,_netdev,auto    0       0"
SEARCH_DOMAIN="xxxx.group xx.xxxxx.ee"

[E0:D5:5E:C1:7C:6C]
HOSTNAME=g2brix1
AUTOLOGIN=tooreg
RELOGIN=0
PASSWORDS_G2BRIX1="pass/_base64_pass"
`

**Comment by alkisg** [comment]
Which display manager are you using?
Login manually to the LTSP client, then paste here whichever of the following files exist:

- /etc/gdm3/daemon.conf
- /etc/gdm3/custom.conf
- /etc/lightdm/lightdm.conf
- /etc/sddm.conf
- /etc/lightdm/slick-greeter.conf

P.S. for multi-line content on github you may use triple-backticks, e.g.

````md
```
content goes here
```
````

**Comment by Valiumlord** [comment]
The server and client use this
/etc/gdm3/custom.conf

```
# GDM configuration storage
#
# See /usr/share/gdm/gdm.schemas for a list of available options.

[daemon]
# Uncomment the line below to force the login screen to use Xorg
#WaylandEnable=false

# Enabling automatic login
#  AutomaticLoginEnable = true
#  AutomaticLogin = user1

# Enabling timed login
#  TimedLoginEnable = true
#  TimedLogin = user1
#  TimedLoginDelay = 10

[security]

[xdmcp]

[chooser]

[debug]
# Uncomment the line below to turn on debugging
# More verbose logs
# Additionally lets the X server dump core if it crashes
#Enable=true

```

**Comment by alkisg** [comment]
LTSP is supposed to automatically modify that file and add the autologin information in it.
Are you sure that what you pasted comes from the client?
We don't care about the server configuration file.

**Reply by Valiumlord** [reply]
sorry, i was on the wrong thing.
so this is what my client one looks like.
/etc/gdm3/custom.conf

```
# GDM configuration storage
#
# See /usr/share/gdm/gdm.schemas for a list of available options.

[daemon]
# Uncomment the line below to force the login screen to use Xorg
#WaylandEnable=false

# Enabling automatic login
#  AutomaticLoginEnable = true
#  AutomaticLogin = user1

# Enabling timed login
#  TimedLoginEnable = true
#  TimedLogin = user1
#  TimedLoginDelay = 10

[security]

[xdmcp]

[chooser]

[debug]
# Uncomment the line below to turn on debugging
# More verbose logs
# Additionally lets the X server dump core if it crashes
#Enable=true


# Generated by `ltsp init`, see man:ltsp(8)
# You can append content here by specifying the GDM3_CONF parameter
[daemon]
AutomaticLoginEnable=false
AutomaticLogin=
```

**Comment by alkisg** [comment]
Get a root shell on the client (e.g. with sudo -i or epoptes) and paste the output of the following command:

```shell
grep tooreg /etc/{passwd,group,shadow}
```

Normally you shouldn't be providing hashes from /etc/shadow on github, but in this case you already pasted the PASSWORDS_ line so it's not a problem.

**Reply by Valiumlord** [reply]
Im having a slight problem with root access. If i do sudo -i and provide with password i get : tooreg is not allowed to run sudo on g2brix1.  This incident will be reported.

**Comment by alkisg** [comment]
Run `pkexec bash`, it'll prompt you with "any sudoer" instead of tooreg.

If that doesn't work, add tooreg in the sudo group on the server, run `ltsp initrd` and reboot the client, so that tooreg will be able to use sudo.

**Reply by Valiumlord** [reply]
Neither of those work on client a.k.a no access to root. 
I added tooreg to sudo group (server shows that its in sudo group but client side only shows "domain users")
pkexec bash ask me the account "ltsp" password which i also changed in the server but it still says wrong password.
(I'm sorry, im kinda new to LTSP or working with Linux, started this February)
Note: tooreg isnt local account, its from NFS side.

**Comment by alkisg** [comment]
> Note: tooreg isnt local account, its from NFS side.

NFS doesn't provide user accounts, I guess you mean something like LDAP or SSSD or AD.
AUTOLOGIN currently doesn't work with LDAP users, see the related issue and discussions:
https://github.com/ltsp/ltsp/issues/399
https://github.com/ltsp/ltsp/discussions/312

I'll work on that in the future when I get some free time, in the meantime patches welcome :)

**Reply by Valiumlord** [reply]
Yes, its SSSD.
Ah, okay. 
Thanks for the answers :)
